interfaces/builtin/yubikey.go

```diff
@@ -0,0 +1,123 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/udev"
+)
+
+const yubikeySummary = `allows access to yubikey devices`
+
+const yubikeyBaseDeclarationSlots = `
+  yubikey:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+type yubikeyDevice struct {
+	VendorId, ProductId string
+}
+
+// https://github.com/Yubico/libu2f-host/blob/master/70-u2f.rules
+var yubikeyDevices = map[string]yubikeyDevice{
```

Comment:
Is there a need for this to be a map? I would think that it is more natural to have a list of things like:
```
type yubikeyInfo struct {
   VendorIDPattern, ProductIDPattern string
   Name string
}
```

Reply:
Done